### PR TITLE
fix: Added decimal.Decimal to _types.py.

### PR DESCRIPTION
**Pull request by ericf-firebolt**
As stated, added `decimal.Decimal` to `_types.py` to support dbt.

### CODE REVIEW

File: src/firebolt/async_db/_types.py [hunk 1]
```diff
@@ -177,7 +178,7 @@ def parse_value(
     """Provided raw value and python type, parses first into python value."""
     if value is None:
         return None
-    if ctype in (int, str, float):
+    if ctype in (int, str, float, Decimal):
```
Comment (stepansergeevitch) [inline src/firebolt/async_db/_types.py:181]: I think that this condition is redundant.
We won't ever have ctype `Decimal`

**Review by stepansergeevitch — COMMENTED** [review]
Please also add unit tests

File: src/firebolt/async_db/_types.py (hunk 1 above)
Comment (ericf-firebolt) [reply src/firebolt/async_db/_types.py]: Done.

**Comment by ericf-firebolt** [comment]
> Please also add unit tests

Decimal is a little weird/annoying in its inconsistency when converting numerical types vs strings, so I added two checks, one sending in a str, which returns a rather long string, and one sending in a numerical type—in this case a float—which returns a string with the same "significant digits" as the number that was input.

**Review by stepansergeevitch — APPROVED** [review]

**Pull request merged**